src/Admin/Polyfills.php

```diff
@@ -0,0 +1,130 @@
+<?php
+/**
+ * Handles backward compatibility of assets for older versions of WP.
+ *
+ * @since 2.0
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP\Admin;
+
+use AmpProject\AmpWP\Infrastructure\Registerable;
+use AmpProject\AmpWP\Infrastructure\Service;
+use WP_Scripts;
+use WP_Styles;
+
+/**
+ * Registers assets that may not be available in the current site's version of core.
+ *
+ * @since 2.0
+ */
+final class Polyfills implements Service, Registerable {
```

Comment:
I made an effort to make this class `Delayed` and `Conditional` but there were several moving parts involving both the admin and paired, and some paired-mode tests were failing. I could tell it was going to take me a while to work through it, so instead we're running this all the time. There are still conditionals, though, to only register assets if they are not already registered in core. Open to suggestions for handling this better.

Reply:
I'd say this is one the few services where it needs to always be loaded. As for delaying it, I'm not sure which action would be most appropriate to use; I'd defer to @westonruter.

Reply:
If the `Conditional` is too complex to use from here, it could potentially be turned around. We could say it'd only register on `'amp_register_polyfills'` and then `do_action( 'amp_register_polyfills' )`  in the parts of the code where we need them.

Not making this Conditional would mean we load it for other plugins as well, and could for example break if another plugin hard-codes shims on its own settings page.

Reply:
Thanks for this idea. Done in c955d3a8e8edd0118202b994dc1ff4561719d7d2